src/main/java/com/gabrielpolak/ticket/Model/DAO/Ticket.java

```
@@ -0,0 +1,64 @@
+package com.gabrielpolak.ticket.Model.DAO;
+
+import com.gabrielpolak.ticket.Model.Request.TicketRequest;
+import com.gabrielpolak.ticket.TicketProperties;
+import com.gabrielpolak.ticket.TicketType;
+import lombok.Data;
+
+import javax.persistence.*;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Data
+@Entity
+public class Ticket {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+    private BigDecimal price;
+    private TicketType type;
+
+    public Ticket() {
+
+    }
+
+    public static Ticket createNewTicket(TicketType type){
+        return new Ticket(type);
+    }
+
+    public Ticket(TicketType type){
+        this.type = type;
+        this.price = getTicketPrice(type);
+    }
+
+    private BigDecimal getTicketPrice(TicketType type){
+        Map<String, String> prices = TicketProperties.getTicketPrices();
+        switch (type){
+            case Adult -> {
+                return new BigDecimal(prices.get("ADULT"));
+            }
+            case Child -> {
+                return new BigDecimal(prices.get("CHILD"));
+            }
+            case Student -> {
+                return new BigDecimal(prices.get("STUDENT"));
+            }
+        }
+        return new BigDecimal(0);
```

Imo for unsupported types we should return an exception instead of null

---

Fixed